### PR TITLE
Force storyboard's loop commands to be run at least once

### PR DESCRIPTION
**Pull request by Wieku**
It looks like in stable storyboard's loop command is forced to be run at least once, which is visible in recently qualified [mapset](https://osu.ppy.sh/beatmapsets/1388906#osu/2868388) that contains a lot of loop commands with loop count set to 0:
```
Sprite,Background,TopCentre,"SB\be\black.png",320,240
 L,1002,0
  ...
 L,11973,0
  ...
 L,22259,0
  ...
 L,23630,0
  ...
 L,25002,0
  ...
 L,26361,0
  ...
 L,29116,0
  ...
 L,30487,0
  ...
 F,0,487,659,0,0.9
 M,0,659,,384.1058,60.37687
 F,0,659,,0.9
 R,0,659,2030,0.2,0.3
 C,0,1002,,0,0,0
 M,0,11630,11730,320,240,384.1058,60.37687
 R,0,11630,11730,0,0.2
 R,0,11730,13002,0.2,0.3
 M,0,32030,32130,200,240,320,240
 R,0,32030,32130,1.83,0
 F,0,33402,,0
```

Storyboard in stable behaves normally, but in lazer it looked like this:
osu_2021-09-29_01-06-19

After adding that cap it looks correct now:
osu_2021-09-29_01-07-43

### CODE REVIEW

**Comment by peppy** [comment]
That sounds so broken. We might want to fix this at stable instead? Does that mean loops play at least twice now regardless?

**Comment by frenzibyte** [comment]
From reading the code, this PR enforces command loops to iterate at least once (not twice). It sounds good to fix this stable-side but I'm a bit unsure whether that linked beatmap is the only one using this quirk.

I think a better way would be to handle this at `LegacyStoryboardDecoder` with an inline comment attached, rather than at `CommandLoop`.

**Comment by peppy** [comment]
Okay, if loop=1 means no loops (1 invocation) then it's more sane. Probably need some better commenting surrounding it though.

**Comment by Wieku** [comment]
Yeah, I'll move it to legacy decoder later today/tomorrow.
My guess it's probably some sort of a command grouping utility in a 3rd party storyboard editor so there's a high chance other storyboards have 0 count loops as well.

**Comment by peppy** [comment]
It might be fine to keep it non-legacy, and rename it to `repeatCount` or something? I'm not sure if a loop that never plays should ever be a thing (likely why stable is doing what it's doing)

**Comment by peppy** [comment]
I've renamed the parameter to hopefully read much better and added guards all over the place. We don't really have storyboard command tests in place yet and I don't know if this is the right time to put in the effort, but the chance of breaking seems low enough that I'm happy with this as-is.

Requesting at least one more review since I made considerable changes to the flow.

**Review by peppy — DISMISSED** [review]

File: osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs [hunk 1]
```diff
@@ -176,8 +176,8 @@ private void handleEvents(string line)
                     case "L":
                     {
                         var startTime = Parsing.ParseDouble(split[1]);
-                        var loopCount = Parsing.ParseInt(split[2]);
-                        timelineGroup = storyboardSprite?.AddLoop(startTime, loopCount);
+                        var repeatCount = Parsing.ParseInt(split[2]);
+                        timelineGroup = storyboardSprite?.AddLoop(startTime, Math.Max(0, repeatCount));
```
Comment (Wieku) [inline osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs:180]: I'm pretty sure we need to subtract 1 from `repeatCount` here as `L,time,1` will make the legacy loop run two times, not one.

Comment (peppy) [reply osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs:180]: Good catch

**Review by peppy — DISMISSED** [review]

**Comment by bdach** [comment]
I've taken the liberty of adding a test case for this. Please check whether it matches your expectations.

**Review by bdach — APPROVED** [review]

**Pull request merged**